packages/fetchai/skills/tac_control_contract/handlers.py

```diff
@@ -0,0 +1,427 @@
+        """
+        Implement the reaction to a message.
+
+        :param message: the message
+        :return: None
+        """
+        oef_message = cast(OEFMessage, message)
+        oef_type = oef_message.type
+
+        self.context.logger.debug(
+            "[{}]: Handling OEF message. type={}".format(
+                self.context.agent_name, oef_type
+            )
+        )
+        if oef_type == OEFMessage.Type.OEF_ERROR:
+            self._on_oef_error(oef_message)
+        elif oef_type == OEFMessage.Type.DIALOGUE_ERROR:
+            self._on_dialogue_error(oef_message)
+        else:
+            self.context.logger.warning(
+                "[{}]: OEF Message type not recognized.".format(self.context.agent_name)
+            )
+
+    def _on_oef_error(self, oef_error: OEFMessage) -> None:
+        """
+        Handle an OEF error message.
+
+        :param oef_error: the oef error
+
+        :return: None
+        """
+        self.context.logger.error(
+            "[{}]: Received OEF error: answer_id={}, operation={}".format(
+                self.context.agent_name, oef_error.id, oef_error.operation
+            )
+        )
+
+    def _on_dialogue_error(self, dialogue_error: OEFMessage) -> None:
+        """
+        Handle a dialogue error message.
+
+        :param dialogue_error: the dialogue error message
+
+        :return: None
+        """
+        self.context.logger.error(
+            "[{}]: Received Dialogue error: answer_id={}, dialogue_id={}, origin={}".format(
+                self.context.agent_name,
+                dialogue_error.id,
+                dialogue_error.dialogue_id,
+                dialogue_error.origin,
+            )
+        )
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+
+class TransactionHandler(Handler):
+    """Implement the transaction handler."""
+
+    SUPPORTED_PROTOCOL = TransactionMessage.protocol_id  # type: Optional[ProtocolId]
+
+    def setup(self) -> None:
+        """Implement the setup for the handler."""
+        pass
+
+    def handle(self, message: Message) -> None:
+        """
+        Implement the reaction to a message.
+
+        :param message: the message
+        :return: None
+        """
+        tx_msg_response = cast(TransactionMessage, message)
+        contract = self.context.contracts.erc1155
+        ledger_api = cast(LedgerApi, self.context.ledger_apis.apis.get("ethereum"))
+        if tx_msg_response.tx_id == "contract_deploy":
+            tx_signed = tx_msg_response.signed_payload.get("tx_signed")
+            tx_digest = ledger_api.send_signed_transaction(
+                is_waiting_for_confirmation=True, tx_signed=tx_signed
+            )
+            transaction = ledger_api.get_transaction_status(  # type: ignore
+                tx_digest=tx_digest
+            )
+            self.context.logger.info(transaction)
```

Comment:
Please improve the logger statement. distinguishes between a successful an unsuccessful case via control flow here.

Reply:
In the unsuccessful case the skill should stop running: deregister itself `self.context.is_active = False`.